### DISCUSSION TITLE
Zweiter Post Button im neuen Log Form

### DISCUSSION THREAD

**Discussion by ChristianGK-GC** [2Abendsegler/GClh#2237] [Ideas]
Was haltet ihr davon, wenn wir einen zweiten Post Button im neuen Log Form nochmal oben anzeigen.
Ich finde es immer etwas umständlich über die TBs zu scrollen um das Log abschicken zu können.
Beispiel:
![grafik](https://user-images.githubusercontent.com/24353480/195782140-9f906236-d4f9-4ed4-a8b6-51b9632e17bb.png)

Code:

    if (document.location.href.match(/\.com\/play\/geocache\/gc\w+\/log/)) {
        try {
            var button = document.getElementById('submitLog');
            var newButton = button.cloneNode(true);
            document.getElementById('logAttachments').after(newButton);
        } catch(e) {gclh_error("second submitButton on new logform",e);}
    }

**Comment by 2Abendsegler** [comment]
Moin,

es erscheint mir schwierig ihn verlässlich mit vollem Funktionsumfang aufzubauen. Wir hatten es nicht geschafft die F2 Funktionalität stabil zu etablieren, weil der Button quasi machte was er wollte und unsere Funktionalität mal da war und mal nicht, auch wenn das schon eine Weile her ist. Und nun wollen wir parallel einen neuen Button aufbauen. Ich glaube nicht, dass mit dem Coding beide Buttons korrekt und synchron funktionieren werden.

Hast du schon ein paar Tests gemacht. Log abschickt, Log abschickt ohne Text, Bilder hochgeladen, Log abschickt mit Bildern, oder so?

**Reply by ChristianGK-GC** [reply]
Hab erst mal `settings_unsaved_log_message` ausgeschalten.
Mit Bilderupload und Co hatte ich keine Probleme.
Ich hatte nur einmal ein Problem mit dem unteren Button. Beim zweiten Versuch ging es dann... Habs nicht verstanden.
Ich teste einfach noch weiter rum.

**Reply by 2Abendsegler** [reply]
Ich kann mir ehrlich gesagt einen zweiten Button nicht so recht vorstellen. Nur um nicht ausschließlich mit meiner Vorstellungskraft zu argumentieren, habe ich auch mal kurz angetestet. Bei Hochladen von Bilder wird nur der neue, erste Button deaktiviert, der original, zweite Button aber nicht. Insofern hm ...

**Comment by 2Abendsegler** [comment]
... was mir gerade noch einfällt. Es gibt den Parameter [Auto open trackable inventory](https://www.geocaching.com/my/#GClhShowConfig#a#settings_auto_open_tb_inventory_list), mit dem der Trackable Bereich immer aufgeklappt bleibt. Vermutlich ist das bei dir so eingestellt. Wenn man den deaktiviert, behält sich der Parameter den letzten Stand. Du kannst ihn also einmal einklappen und weg ist er, bis du ihn wieder aufklappst.

**Reply by ChristianGK-GC** [reply]
Das habe ich auch entdeckt und half mir etwas, wobei ich am Ende dann doch immer den näher gelegenen Button verwendet hab.

**Reply by capoaira** [reply]
Oh, den kannte ich gar nicht. Mich nervt das immer offene TB-Menü und damit verbundene Scrollen auch immer. Vielleicht sollten wir den Parameter default auf `false` setzten. Vermutlich sind wir nicht die einzigen, nur die meisten halten es für eine Funktion von GS, bei der man sich fragt "Is it a bug, or a feature?" 😅

**Reply by 2Abendsegler** [reply]
> Das habe ich auch entdeckt und half mir etwas, wobei ich am Ende dann doch immer den näher gelegenen Button verwendet hab.

Eigentlich ist der Parameter mittlerweile fast hinfällig weil GS sich die Stellung merkt und es beim nächsten mal wieder genauso setzt, sofern die Cookies erhalten bleiben. Wenn wir ihn behalten wäre false wohl besser, sehe ich auch so. Wollen wir ihn überhaupt behalten? Vermutlich sorgt er tatsächlich für Unmut. Oder wir setzen ihn auf false und zusätzlich einmalig bei allen Usern auf false.

**Reply by capoaira** [reply]
>Eigentlich ist der Parameter mittlerweile fast hinfällig weil GS sich die Stellung merkt und es beim nächsten mal wieder genauso setzt, sofern die Cookies erhalten bleiben.

Sehe ich auch so, dann muss man die halt alle paar Monate/Jahre auf bzw. zu klappen, je nachdem was der Default Wert bei GS ist. Ich denke sowieso, dass das das Intuitivste ist.

Dann wäre meiner Meinung auch dieser Vorschlag hinfällig, da man nicht immer scrollen müsste ;)

**Reply by 2Abendsegler** [reply]
Gut, ich werde den Parameter [Auto open trackable inventory (settings_auto_open_tb_inventory_list)](https://www.geocaching.com/my/#GClhShowConfig#a#settings_auto_open_tb_inventory_list) entfernen.

> Dann wäre meiner Meinung auch dieser Vorschlag hinfällig, da man nicht immer scrollen müsste ;)

Ja, sehe ich auch so. 
@ChristianGK-GC 
sorry

**Reply by ChristianGK-GC** [reply]
Passt. Mach's zu.
Wenigstens hat es zu einer Verbesserung in GCLH geführt.
